jbmc/src/java_bytecode/java_types.cpp

```diff
@@ -171,9 +169,9 @@ bool is_multidim_java_array_type(const typet &type)
 /// \param tag Tag of a struct
 /// \return True if the given string is a Java array tag, i.e., has a prefix
 /// of java::array[
-bool is_java_array_tag(const irep_idt& tag)
+bool is_java_array_tag(const symbol_typet &tag)
```

Comment:
How does this strengthen the check? I think if you want to improve this, what we've been meaning to do for a while is introduce a `java_array_object_typet` (name up for discussion) that encapsulates all of these `java_array` methods a bit.